### ISSUE TITLE
pushing commit from shippable to github just failed silently?

### ISSUE THREAD

**Issue by osumampouw** [Shippable/support#1669]
Hello,

I have a use case that should pretty similar to a lot of other people. 

Once my build is successful, I will bump the version of package.json and then I will commit and push back to github. 
---
Turns out this is not possible. So I followed the tutorial here: http://avoidingneighbours.blogspot.com/2015/03/pushing-changes-from-shippable-to-github.html
http://blog.shippable.com/demystifying-shippables-deployment-keys
which basically told me to use the deployment key in shippable project setting and adding it to github's deploy-keys (making sure the write permission is turned on). And then, do the following in the "before_script"
```
  - git remote set-url origin git@github.com:$REPO_NAME.git
  - git config --get remote.origin.url
```

The bump for package.json happens in after_success, you can see my shippable script is like this:
```
before_script:
  - git remote set-url origin git@github.com:$REPO_NAME.git
  - git config --get remote.origin.url
  - eval "$(ssh-agent -s)"
  - ssh-add ~/.ssh/id_rsa # I added this because I notice the ssh-agent is empty in shippable
  - ssh-add -l # making sure the ssh-agent is there
  - npm install -g gulp
script:
  - gulp build
after_success:
  - python slack_notifier.py --slack-url $SLACK_URL --message success
  - git remote set-url origin git@github.com:$REPO_NAME.git
  - git config --get remote.origin.url
  - eval "$(ssh-agent -s)"
  - ssh-add ~/.ssh/id_rsa # do the same as above, just to make sure the ssh-agent is still there before we push back to github
  - ssh-add -l
  - if [ "$BRANCH" = "production" ]; then gulp ciTag; fi
  - if [ "$BRANCH" = "master" ]; then gulp ciBumpRequireJS; fi
```
When I run the script:
the build succeeds but the git push doesn't work. And what's strange is there is no error. This is the output of my gulp ciBumpRequireJs
```
if [ "$BRANCH" = "master" ]; then gulp ciBumpRequireJS; fi
[08:48:51] Warning: gulp version mismatch:
[08:48:51] Global gulp is 3.9.0
[08:48:51] Local gulp is 3.8.11
[08:48:55] Using gulpfile ~/workspace/src/github.com/cermati/rajamoney/gulpfile.js
[08:48:55] Starting 'ciSafeguard'...
[08:48:55] Finished 'ciSafeguard' after 108 μs
[08:48:55] Starting 'ciGitPrepare'...
[08:48:55] Your branch is up-to-date with 'origin/master'.
Already on 'master'
[08:48:55] git config --global user.name cermati-robot (log : false)
[08:48:55] git config --global user.email robot@cermati.com (log : false)
[08:48:55] git remote set-url origin https://cermati-robot:CerdasCermat1@github.com/cermati/rajamoney.git (log : false)
[08:48:55] Finished 'ciGitPrepare' after 185 ms
[08:48:55] Starting 'ciBumpRequireJS'...
[08:48:55] git log -n 1 --pretty=format:"%ce" (log : false)
AAAAA Start for branch master
AAAAA Done
[08:48:55] Bumped 'package.json' requirejs_build to: 0.1.3
[08:48:55]
[08:48:55] [master 344e662] Auto-bump requireJS [ci skip]
1 file changed, 1 insertion(+), 1 deletion(-)
[08:48:55] Everything up-to-date
pushing has no error
[08:48:55] Finished 'ciBumpRequireJS' after 472 ms
 if [ "$BRANCH" != "master" ]; then gulp images; fi
```
here's my gulp task for bumping version:
```
// This task should only be run by inside ci environment
gulp.task('ciBumpRequireJS', ['ciGitPrepare'], function (callback) {
  git.exec({args: 'log -n 1 --pretty=format:"%ce"'}, function (_, stdout) {
    // Prevent bumping if this is a commit by travis
    if (stdout === process.env.ROBOT_EMAIL) {
      return callback();
    }
    async.series([
      // Bump the patch version of the requirejs_build
      function (cb) {
        gulp.src(['./package.json'])
          .pipe(bump({key: 'requirejs_build'}))
          .pipe(gulp.dest('./'))
          .pipe(git.add())
          .pipe(git.commit('Auto-bump requireJS [ci skip]')).on('error', function(err){
            console.log("error git", err);
          });
        cb();
      },
      // Push changes
      function (cb) {
        console.log('AAAAA Start for branch ' + process.env.BRANCH);
        git.push('origin', process.env.BRANCH, function(err){
          if (err){
            console.log("error pushing", err);
          } else {
            console.log("pushing has no error");
          }
          cb(err);
        });
        console.log('AAAAA Done');
      }
    ], callback);
  });
});
```
As you can see, if there's an error thrown in git.push(), it should be printed to console, but I don't get any error. So that means the git push is successful. Yet after the build succeeds, I still don't see the push being reflected in my "master" branch. I don't see any commit that says "Auto-bump requireJS [ci skip]"

Can I get some help? Maybe there's something that I forgot to do? 

PS: for your reference this is the link to the build https://app.shippable.com/builds/559f864b72c8990c00324db7

**Comment by rageshkrishna** [comment]
Hi @osumampouw! I don't use gulp a lot so I can't really say exactly why gulp is not reporting an error in this case, but I do have a project that does exactly what you're trying to do (bump package.json, push to GitHub and publish to npm): https://github.com/rageshkrishna/shippable.js/

If you look at [my YML file[](https://github.com/rageshkrishna/shippable.js/blob/master/.shippable.yml) you'll see that i'm able to do all this with just one line in my `after_success`:

````
 - if [ "$BRANCH" == "master" ] && [ "$PULL_REQUEST" == false ]; then npm version patch && git commit --amend -m "Version bump [skip ci]" && git push git@github.com:rageshkrishna/shippable.js.git master && npm publish; fi;
````

I added the deployment key from Shippable to my GitHub account to get the git push working. This is really the only requirement to get the authentication working for pushing to the repository.

Here's the resulting commit from a build I ran 10 minutes ago that shows just the version number being bumped up by `npm version patch`: https://github.com/rageshkrishna/shippable.js/commit/f1a090586c5120309839682e441e88e2d6d4e40b
